### DISCUSSION TITLE
ソースコードについての質問

### DISCUSSION THREAD

**Discussion by nescokubo** [FisaCloudAlliance/JamstackEducation#63] [Q&A]
お疲れ様です。
今回頂いたリポジトリの「src\store\IActionBase.ts」の3行目（[prop: string]: any;）の使用例や、どういった理由でこの記述になっているのかを知りたいです。
私個人でも調べてみたのですが、中々腑に落ちる情報がなく、どうしても理解できなかったので質問させていただきました。
React は redux を使った簡単な計算機などを作ったことがあるくらいの理解度です。
お忙しいところ恐縮ですが、ご回答よろしくお願いいたします。

`export interface IActionBase {
    type: string;
    [prop: string]: any;
}`

**Comment by s-ueno** [comment]
とても良いご質問、ありがとうございます！

ご質問の背景は二つあると存じます。

+ `[prop: string]: any;` という TypeScript 構文
+ `IActionBase` を利用した redux の仕組み

こちら、それぞれにWikiおよびサンプルソースを main リポジトリ に用意いたしました。

前者については [TypeScriptのWiki ](https://github.com/FisaCloudAlliance/JamstackEducation/wiki/70-Typescript)をご参照ください。

後者については[react のWiki](https://github.com/FisaCloudAlliance/JamstackEducation/wiki/80-react-hooks)をご参照ください。

また、サンプルのソースは main から 新しいブランチを切って、ローカルで ブランチのソースをご確認ください。

![image](https://user-images.githubusercontent.com/8575906/145934886-e46599de-9c45-444f-bf1d-28be2933ae80.png)

また、説明が足らないところなどありましたら、気兼ねなくご質問いただければ、幸いです。

他のメンバーにも情報共有できるので助かります。

**Reply by nescokubo** [reply]
迅速に回答いただき、ありがとうございます。
上記二つのWikiを読んだうえで、
`[prop: string]: any;`
に対する私の理解が正しいか確認していただきたく思います。

IActionBase における `[prop: string]: any;` とは、
IActionBase を継承しているオブジェクトの全てのプロパティに、 `hoge["myProperty"]` のように型を意識しなくても型セーフにアクセスすることを可能にするためのものであると考えます。
私の理解に間違い等ありましたら、ご指摘お願いします。

ソースコード全体は確認できていないので、後日また質問することがあるかもしれません。
その際は、よろしくお願いいたします。
